anthos-bm-gcp-terraform/main.tf

```diff
@@ -323,6 +323,16 @@ module "init_hosts" {
   cluster_yaml_path      = local.cluster_yaml_file
 }
 
+module "gke_hub_membership" {
+  source   = "terraform-google-modules/gcloud/google"
+  version  = "~>3.1.1"
+  enabled  = var.mode == "install" || var.mode == "manuallb" ? true : false
+  platform = "linux"
+  # The hub membership is created by the bmctl create cluster step
+  destroy_cmd_body  = "container hub memberships delete --quiet --project ${var.project_id} ${var.abm_cluster_id}"
+  module_depends_on = [module.install_abm]
```

Comment:
Whats the difference between `depends_on` and `module_depends_on`. Can we use `depends_on` to be consistent with the other places?

Reply:
Can we move this to the top under `version`

Reply:
The `module_depends_on` is provided by the module as using the normal `depends_on` will conflict with a `count` in the module that isn't evaluated till apply.  That said, it's not absolutely required and can be removed if desired.  It simply helps to run it at a more logical seeming time.  Let me know what you prefer.

Reply:
I think if we have it enabled always (any mode) then it makes sense to not have it dependent on any modules. Also, the `install` module won't be doing anything during the destroy phase

Also can we have this moved to the bottom of the script.

Reply:
👍